datadog_checks_downloader/datadog_checks/downloader/download.py

```diff
@@ -125,7 +126,7 @@ def __download_in_toto_root_layout(self):
         # expected version of the root layout. This is so that, for example, we
         # can introduce new parameters w/o breaking old downloaders that don't
         # know how to substitute them.
-        target_relpath = os.path.join(IN_TOTO_METADATA_DIR, self.__root_layout)
+        target_relpath = f'{IN_TOTO_METADATA_DIR}/{self.__root_layout}'
```

Comment:
I'm hesitant to mention Windows there, I'm not sure the comment would make sense without the context of what was there originally. But perhaps it's worth mentioning in https://github.com/DataDog/integrations-core/blob/c73578c33eef65a831957156aa43fc47674b8391/datadog_checks_downloader/datadog_checks/downloader/download.py#L103
to remind ourselves that `target_relpath` is supposesd to be a [path-relative-URL string](https://url.spec.whatwg.org/#path-relative-url-string).

Reply:
But what about other relative paths for TUF metadata (e.g., in `__download_with_tuf()`)?

Reply:
I added a utility here https://github.com/DataDog/integrations-core/pull/13631/commits/847199ab8bb02eec6d9ab31edbc8e7041829c8c3

What else should use that exactly?

Reply:
Thanks! A few comments:
1. I think that utility is perhaps better coupled with the `TUFDownloader` class itself, since nothing else would probably use it.
2. Everything that ends with a `_relpath` and that is used to query TUF (but not for reading what is on disk).

Reply:
I've added a change that ensures that `Updater.get_targetinfo()` always gets called with a path that uses `/` as the separator, converting from any OS-specific path to that.